hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/dog/RoutineEnvironmentTransformations.scala

```diff
@@ -0,0 +1,414 @@
+package org.broadinstitute.monster.dap.dog
+
+import org.broadinstitute.monster.dap.RawRecord
+import org.broadinstitute.monster.dogaging.jadeschema.fragment.HlesDogRoutineEnvironment
+
+object RoutineEnvironmentTransformations {
+
+  /**
+    * Parse all routine environment fields out of a raw RedCap record,
+    * injecting them into a partially-modeled dog record.
+    */
+  def mapRoutineEnvironment(rawRecord: RawRecord): HlesDogRoutineEnvironment = {
+    val init = HlesDogRoutineEnvironment.init()
+
+    val transformations = List(
+      mapDogparks _,
+      mapRecreationalSpaces _,
+      mapWork _,
+      mapSitter _,
+      mapEatsFeces _,
+      mapToys _,
+      mapSleepLocation _,
+      mapToxinsIngested _,
+      mapOtherAnimals _,
+      mapGeneralRoutine _
+    )
+
+    transformations.foldLeft(init)((acc, f) => f(rawRecord, acc))
+  }
+
+  def calculateTotalTime(
+    hoursInput: Option[Long],
+    minutesInput: Option[Long]
+  ): Option[Long] = {
+    for {
+      hours <- hoursInput
+      minutes <- minutesInput
+    } yield {
+      hours * 60 + minutes
+    }
+  }
+
+  /* Dog Parks */
+  def mapDogparks(
+    rawRecord: RawRecord,
+    dog: HlesDogRoutineEnvironment
+  ): HlesDogRoutineEnvironment = {
+    val dogPark = rawRecord.getOptionalBoolean("de_dogpark_yn")
+    // ONLY if: de_dogpark_yn = 1
+    if (dogPark.contains(true)) {
+      val dogparkTravel = rawRecord.get("de_dogpark_get_to")
+      val dogparkTravelOther = dogparkTravel.map(_.contains("98"))
+      dog.copy(
+        deDogpark = dogPark,
+        deDogparkDaysPerMonth = rawRecord.getOptionalNumber("de_dogpark_freq"),
+        deDogparkTravelWalk = dogparkTravel.map(_.contains("1")),
+        deDogparkTravelDrive = dogparkTravel.map(_.contains("2")),
+        deDogparkTravelBike = dogparkTravel.map(_.contains("3")),
+        deDogparkTravelPublicTransportation = dogparkTravel.map(_.contains("4")),
+        deDogparkTravelOther = dogparkTravelOther,
+        //ONLY if: de_dogpark_get_to(98) = '1'
+        deDogparkTravelOtherDescription = if (dogparkTravelOther.contains(true)) {
+          rawRecord.getOptional("de_dogpark_get_to_other")
+        } else None,
+        // Compute using hour & min columns, Multiply hours by 60 to get minutes
+        deDogparkTravelTimeMinutes = for {
+          hours <- rawRecord.getOptionalNumber("de_dogpark_hr")
```

Comment:
In other places, we changed the schema to store hour and minute fields separately because the "hour" fields don't allow the user to enter an integer themselves, they have to select from 0, 1, 2, ... 11, or "12+" (and it doesn't make sense to convert an option like 12+ into minutes).

I'm not sure who would travel 12+ hours to go to the dog park anyway 🤷‍♀️ but maybe it would be worth changing for the sake of consistency? What do others think?

Reply:
Ergh, fair point... It makes me sad but it makes sense

Reply:
Thanks for the suggestion, Sarah! I expanded the schema to add separate "hours" fields for these cases. I removed the code that was adding the minutes and hours.